plugins/inputs/dmcache/dmcache_linux.go

```diff
@@ -0,0 +1,158 @@
+// +build linux
+
+package dmcache
+
+import (
+	"strconv"
+	"strings"
+
+	"errors"
+
+	"github.com/influxdata/telegraf"
+)
+
+const metricName = "dmcache"
+
+var fieldNames = [...]string{
+	"metadata_used",
+	"metadata_free",
+	"cache_used",
+	"cache_free",
+	"read_hits",
+	"read_misses",
+	"write_hits",
+	"write_misses",
+	"demotions",
+	"promotions",
+	"dirty",
+}
+
+func (c *DMCache) Gather(acc telegraf.Accumulator) error {
+	outputLines, err := c.rawStatus()
+	if err != nil {
+		return err
+	}
+
+	total := make(map[string]interface{})
+
+	for _, s := range outputLines {
+		fields := make(map[string]interface{})
+		data, err := parseDMSetupStatus(s)
+		if err != nil {
+			return err
+		}
+
+		for _, f := range fieldNames {
+			fields[f] = calculateSize(data, f)
+		}
+
+		if c.PerDevice {
+			tags := map[string]string{"device": data["device"].(string)}
+			acc.AddFields(metricName, fields, tags)
+		}
+		aggregateStats(total, fields)
+	}
+
+	acc.AddFields(metricName, total, map[string]string{"device": "all"})
+
+	return nil
+}
+
+func parseDMSetupStatus(line string) (map[string]interface{}, error) {
+	var err error
+	status := make(map[string]interface{})
```

Comment:
But, in this case, function "calculateSize" will requre to use reflection to get value by key.